src/text-editor.js

```diff
@@ -1106,7 +1129,7 @@ class TextEditor {
   setEncoding (encoding) { this.buffer.setEncoding(encoding) }
 
   // Essential: Returns {Boolean} `true` if this editor has been modified.
-  isModified () { return this.buffer.isModified() }
+  isModified () { return this.isReadOnly() ? false : this.buffer.isModified() }
```

Comment:
I don't want this unnecessary guard. If readonly really disallow buffer modification, just returning `buffer.isModified()` should be enough.
This method should be just proxy of `this.buffer.isModified()` for clarity, and remove possibility of inconsistency   when it become `this.isReadonly !== this.buffer.isModifed()`.
Also this breaks https://github.com/t9md/atom-narrow pkg's important feature.

Reply:
This is necessary otherwise it is not possible to create a read-only text editor from some in-memory data without it showing as modified on the tab.

Reply:
then, I want to say hack `this.buffer.isModified` by overriding `buffer.isModified = () => true`.
I feel some anxiety when seeing this fundamental return different answer based on `isReadOnly` status, and that status is unstable(changed by just saving buffer).
I can't believe such method, so just trying to relay on `buffer.isModified` directly.

Reply:
Oh, I wanted to say use `buffer.isModified = () => false` to hide modified icon on tab.